post-deploy didn't use environment variables

Hi! 
Not sure if bug but.. 

So, there is  "env"  : {  ...   } block in every deploy targets in ecosystem.json but even if something is specified there, it doesn't applied to post-depoy command hook. 

Here is use-case: I have post-deployment scripts, i.e migrations which have to be executed before application is started just after deploy. And this scripts  are using same config system which takes db connection credentials from env variables. 

It would be great if env variables from ecosystem will be injected not only for apps but for all hooks too. 

Thank you and sorry for bad english

Similar problem here. `pm2 deploy my_server` wasn't able to start app with nvm (`[PM2][ERROR] NVM is not available in PATH`). I have to `export NVM_DIR=/home/me/.nvm` in post-deploy script. When I execute `pm2 start ecosystem.config.js` from my_server shell NVM_DIR variable works as expected.

--- a/deploy.js
+++ b/deploy.js
@@ -15,11 +15,11 @@ function spawn(hostJSON, args, cb) {
   var shellSyntaxCommand = "echo '" + hostJSON + "' | \"" + __dirname.replace(/\\/g, '/') + "/deploy\" " + args.join(' ');
   var proc = childProcess.spawn('sh', ['-c', shellSyntaxCommand], { stdio: 'inherit' });
 
-  proc.on('error', function(e) {
+  proc.on('error', function (e) {
     return cb(e.stack || e);
   });
 
-  proc.on('close', function(code) {
+  proc.on('close', function (code) {
     if (code == 0) return cb(null, args);
     else return cb(code);
   });
@@ -36,12 +36,12 @@ function spawn(hostJSON, args, cb) {
 function deployForEnv(deploy_conf, env, args, cb) {
   if (!deploy_conf[env]) return cb(env + ' not defined in deploy section');
 
-  var piped_data  = JSON.stringify(deploy_conf[env]);
+  var piped_data = JSON.stringify(deploy_conf[env]);
   var target_conf = JSON.parse(piped_data); //effectively clones the conf
 
-  if(target_conf.ssh_options) {
+  if (target_conf.ssh_options) {
     var ssh_opt = '';
-    if(Array.isArray(target_conf.ssh_options)) {
+    if (Array.isArray(target_conf.ssh_options)) {
       ssh_opt = '-o ' + target_conf.ssh_options.join(' -o ');
     } else {
       ssh_opt = '-o ' + target_conf.ssh_options;
@@ -85,15 +85,15 @@ function deployForEnv(deploy_conf, env, args, cb) {
     target_conf.path = path.resolve(target_conf.path);
 
   if (Array.isArray(target_conf.host)) {
-    async.series(target_conf.host.reduce(function(jobs, host) {
-      jobs.push(function(done) {
+    async.series(target_conf.host.reduce(function (jobs, host) {
+      jobs.push(function (done) {
 
         if (process.env.NODE_ENV !== 'test') {
           console.log('--> on host %s', host.host ? host.host : host);
         }
 
         target_conf.host = host;
-
+        target_conf['post-deploy'] = 'export ' + objectToEnvVars(target_conf.env) + ' && ' + target_conf['post-deploy']
         var custom_data = JSON.stringify(target_conf);
 
         spawn(custom_data, args, done);
@@ -105,15 +105,23 @@ function deployForEnv(deploy_conf, env, args, cb) {
     if (process.env.NODE_ENV !== 'test') {
       console.log('--> on host %s', target_conf.host);
     }
+
+    target_conf['post-deploy'] = 'export ' + objectToEnvVars(target_conf.env) + ' && ' + target_conf['post-deploy']
     spawn(JSON.stringify(target_conf), args, cb);
   }
 
   return false;
 }
 
+function objectToEnvVars(obj) {
+  return Object.keys(obj).map(function (key) {
+    return key.toUpperCase() + '=' + obj[key];
+  }).join(' ')
+}
+
 function run() {
-  var conf    = JSON.parse(fs.readFileSync('app.json'));
-  var args    = process.argv;
+  var conf = JSON.parse(fs.readFileSync('app.json'));
+  var args = process.argv;
 
   if (args.indexOf('deploy') == -1)
     throw new Error('deploy argument not found');
@@ -122,7 +130,7 @@ function run() {
 
   var env = args[0];
 
-  Deploy.deployForEnv(conf.deploy, env, args, function(err, data) {
+  Deploy.deployForEnv(conf.deploy, env, args, function (err, data) {
     console.log(arguments);
   });
 }

--- a/deploy.js
+++ b/deploy.js
@@ -114,7 +114,7 @@ function deployForEnv(deploy_conf, env, args, cb) {
 }
 
 function objectToEnvVars(obj) {
-  return Object.keys(obj).map(function (key) {
+  return !obj ? '' : Object.keys(obj).map(function (key) {
     return key.toUpperCase() + '=' + obj[key];
   }).join(' ')
 }

